### DISCUSSION TITLE
In prometheus.yaml file there is no image of prometheus! I'm wondering without image how we can deploy it!

### DISCUSSION THREAD

**Discussion by javad87** [strimzi/strimzi-kafka-operator#7251] [Q&A]
In prometheus.yaml file there is no image of prometheus! I'm wondering without image how we can deploy it! Am I making a mistake or image must be there and missed out by mistake.

I checked another page (https://grafana.com/docs/grafana-cloud/kubernetes-monitoring/prometheus/prometheus_operator/) and in .yaml file they mentioned image:

I'm using microk8s kubernetes cluster and strimizi kafka cluster....also not enabled prometheous add ons from mikrok8s and trying to deploye prometheous from scratch as mentioned in strimzi guide.

apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  name: prometheus
  labels:
    app: prometheus
spec:
  image: **_quay.io/prometheus/prometheus:v2.22.1_**
  nodeSelector:
    kubernetes.io/os: linux
  replicas: 2
  resources:
    requests:
      memory: 400Mi
  securityContext:
    fsGroup: 2000
    runAsNonRoot: true
    runAsUser: 1000
  serviceAccountName: prometheus
  version: v2.22.1
  serviceMonitorSelector: {}

**Comment by scholzj** [comment]
I would expect that the Prometheus Operator will use its default image when image is not specified. That is what most operators do.

**Reply by javad87** [reply]
> I would expect that the Prometheus Operator will use its default image when image is not specified. That is what most operators do.



I searched for image key word  inside bundle.yaml (https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/main/bundle.yaml) prometheus operator file, the only image found is "image: quay.io/prometheus-operator/prometheus-operator:v0.58.0" which is for Deployment of prometheus-operator, no image for "prometheus" itself to the best of my knowledge!

BTW, I have prometheus and Grafana on another server different from the server I deployed my 'kube&kafka' cluster, I do not want to deploy prometheus and Grafana again on kube server...How can I like node_exporter scrape kubes Pod, services, resources metrics and kakfa_exporter, expose its metric to my remote prom+grafana server? already deployed kafka_exporter, but dont know endpoint is on which port? (9094?)  or for sending /metrics to promo server which ports I should use?
Could you please provide a guide (or video tuturial ) for situation when there exists promo+grafana on remote server and wants to send metrics ( most of enterprises have already prom+grafana for montitoring their resources and it is not wise to deploy it again in kube server! )

tnx for ur reply;

**Reply by scholzj** [reply]
I don't think Strimzi can really support you with Prometheus issues. Strimzi provides the Prometheus endpoints. The installation files are examples only, you are expected to use whatever you want to run Prometheus and run it in any way you want. You just need to scrape the metrics.

Just to be clear, there is nothing wrong with asking about it here. But it might not be the best forum where you find Prometheus experts.

**Reply by javad87** [reply]
tnx @scholzj I understand and know that providing support for strimzi project has great responsibility and time consuming for u guys , I made promo+grafana worked with microk8s cluster and kafka strimzi 3 borkers cluster  by using the page mentioned above and strimzi example .yaml files except adding "promo image"  to .yaml file, I deployed prometheus and port-forward 9090 to get access from remote server with this command:
$ nohup kubectl port-forward prometheus-prometheus-0 9090:9090 -n monitoring --address 0.0.0.0&
I also noticed kafka_exporter sends metrics on port 9094. and we can forward that port inorder to get access from remote promo server.

these are sample kafka_exporter and cruise_control dashboard for clarification:

![image](https://user-images.githubusercontent.com/38490446/187643517-be8210e8-40ea-4cfd-b6c2-580d7a716df1.png)

![image](https://user-images.githubusercontent.com/38490446/187643553-df76fda5-1583-48ea-a337-750f28eb56b6.png)
